pkg/sqlstore/impl/system/migrations/002_receipterroridx.up.sql

```diff
@@ -0,0 +1 @@
+ALTER TABLE system_txn_receipts ADD error_event_idx INTEGER DEFAULT 0;
```

Comment:
Uhm, this default is 100% wrong. The thing here was that at this point in time, I was doubting if making it nullable or giving it a 0 value and not making it null.

The default's main point was to migrate _existing data_. In that case, those existing entries should be `null`, not 0.

I want to keep the fact that the column can be null since I want to optimize for our future API evolution that doesn't have this weird-ish -1 interpretation.

So considering that, I'll:
1) Keep the `INTEGER` that can be null.
2) Remove the `DEFAULT 0`
3) Include an extra SQL query that would do something like `UPDATE ... SET error_event_idx=0 WHERE error is not null`.

The extra point is the last one. LMK if makes sense to you to double check.

Reply:
Totally forgot about the migration of old rows. Now I get your intention. But yeah. the 3 steps make sense!